providers/base/tests/test_stress_ng_test.py

```diff
@@ -0,0 +1,292 @@
+#!/usr/bin/env python3
+
+# Copyright 2024 Canonical Ltd.
+# Written by:
+#   Pedro Avalos Jimenez <pedro.avalosjimenez@canonical.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3,
+# as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import unittest
+from subprocess import CalledProcessError, TimeoutExpired
+from unittest.mock import MagicMock, mock_open, patch
+
+from stress_ng_test import main, num_numa_nodes, swap_space_ok
+
+
+class TestMemoryFunctions(unittest.TestCase):
+    @patch("stress_ng_test.run")
+    def test_num_numa_nodes_success(self, run_mock):
+        run_mock.return_value = MagicMock()
+        run_mock.return_value.stdout = b"available: 2 nodes (0-1)"
+        self.assertEqual(num_numa_nodes(), 2)
+
+    @patch("stress_ng_test.run", side_effect=OSError)
+    def test_num_numa_nodes_failure(self, run_mock):
+        self.assertEqual(num_numa_nodes(), 1)
+
+    @patch(
+        "stress_ng_test.psutil.swap_memory", return_value=MagicMock(total=1)
```

Comment:
In general, I wouldnt mock a sysmodule this way. I don't think (correct me if I'm wrong) that this makes any difference from just `@patch("psutil.swap_memory"...)` but the extra text makes it hard to know its an external module

Reply:
Resolved in 7c57d6d